plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -1930,6 +1930,47 @@ chain::signed_block_ptr read_only::get_block(const read_only::get_block_params&
    return block;
 }
 
+read_only::get_block_header_result read_only::get_block_header(const read_only::get_block_header_params& params, const fc::time_point& deadline) const{
+   std::optional<uint64_t> block_num;
+
+   EOS_ASSERT( !params.block_num_or_id.empty() && params.block_num_or_id.size() <= 64,
+               chain::block_id_type_exception,
+               "Invalid Block number or ID, must be greater than 0 and less than 64 characters"
+   );
+
+   try {
+      block_num = fc::to_uint64(params.block_num_or_id);
+   } catch( ... ) {}
+
+   if (!params.include_extensions) {
+      std::optional<signed_block_header> header;
+
+      if( block_num ) {
+         header = db.fetch_block_header_by_number( *block_num );
```

Comment:
No try block for `db.fetch_block_header_by_number`?

Reply:
The try you saw in the else was thrown from the type conversion from `string` to `block_id_type`, not from the `fetch_block_header_by_id` itself. In this line, the block_num is already the right type; therefore, its not going to throw here. This is exactly the same logic with `get_raw_block()` in line 1920.